### DISCUSSION TITLE
how to use Button to check all checkboxes

### DISCUSSION THREAD

**Discussion by Mohsing511** [livewire/livewire#2324] [1. Help]
I need help I want to use a button (select all) but don't know how to check all checkboxes using a function please guide me. thanks in advance.
```
<button wire:click="selectAll()" class="transform hover:scale-110 border-4 text-xl bg-red-500 hover:bg-red-700">Select All</button>
 <td class="px-6 py-4 whitespace-nowrap">
                            <input type="checkbox" name="checkbox" id="checkbox" value="{{ $staff->id }}" wire:model="recordSelected">{{ $staff->id }}
                        </td>
```
and function is empty so please guide me 
 ```
public function selectAll(){
}

```

**Comment by ju5t** [comment]
@Mohsing511 this is covered in this screencast: https://laravel-livewire.com/screencasts/s7-select-all. It's a sponsor only thing, but that's a small price to pay considering the value you get out of it.

**Reply by Mohsing511** [reply]
yes, buddy, I know that but believe me, I have not much money I have only $5 in my Payoneer account and that's all I have so I need free help. please guide me I am just a beginner :(.

**Reply by Wulfheart** [reply]
Consider writing a message to @calebporzio and explain your situation. I think he can help you with this regarding discounts/purchase parity.

**Comment by joshhanley** [comment]
@Mohsing511 I agree with @ju5t that the screencasts are worth their weight in gold. But I do understand sometimes it's not possible to pay.

What I would suggest is trying to write your `selectAll` method yourself, and then posting up what you tried and any errors you have, so people can help you debug. 

That way you get the benefit of trying to learn how to do it, and if you run into any issues, then people have something more concrete to help you with. You are less likely to get help if it's just asking people to write it for you.

You write it and then we can help you fix it 🙂

I would also recommend sharing more code. For example: it looks like you have access to a `$staff` variable. Where does that come from? If that's in a loop, it's worth sharing the loop too and any surrounding code that provides more context.

Hope this helps!
